include/libtorrent/torrent_handle.hpp

```diff
@@ -693,7 +701,6 @@ namespace aux {
 		//	ses.pause();
 		//	for (torrent_handle const& h : handles)
 		//	{
-		//		if (!h.is_valid()) continue;
```

Comment:
the problem with removing this is that using an invalid handle later will throw an exception. Technically, this is a problem with the existing code too, since the handle may become invalid at any time. But maybe a better way to do this would be to add a try-catch block to the for-loop, like this:

```
for (torrent_handle const& h : handles) try
{
   torrent_status s = h.status();
   if (!s.has_metadata || !s.need_save_resume_data()) continue;
   ...
}
catch (lt::system_error const& e)
{
   // the handle was invalid, ignore this one and move on to the next
}
```